### DISCUSSION TITLE
Wasm support

### DISCUSSION THREAD

**Discussion by aluedeke** [nats-io/nats.rs#1045] [Ideas]
I would love to use nats.rs inside a wasm. Are there any plans to support this (e.g. nats over websocket in wasm).

**Comment by caspervonb** [comment]
Hey there.

> I would love to use nats.rs inside a wasm.

Multiple layers to what it means to support WebAssembly, so let's start with that! In which runtime environment?

It's not just the transport that has to be supported, the threading model is different. Tokio has some support for wasm32-wasi but there are things that are going to be missing because the required system calls are not in WASI.

Not completely impossible but it would be with a lot of caveats.

**Reply by aluedeke** [reply]
Hey,

it should run inside browsers. I am using [Yew](https://github.com/yewstack/yew) to develop my frontend and would love to use nats to connect to the backend.
